services/navigation/builtin/builtin.go

```diff
@@ -622,7 +637,154 @@ func (svc *builtIn) waypointIsDeleted() bool {
+		if err != nil {
+			return nil, err
+		}
+
+		// determine transform from camera to movement sensor
+		movementsensorOrigin := referenceframe.NewPoseInFrame(svc.movementSensor.Name().ShortName(), spatialmath.NewZeroPose())
+		cameraToMovementsensor, err := svc.fsService.TransformPose(ctx, movementsensorOrigin, detector.CameraName.ShortName(), nil)
+		if err != nil {
+			// here we make the assumption the movementsensor is coincident with the camera
+			svc.logger.Debugf(
+				"we assume the movementsensor named: %s is coincident with the camera named: %s due to err: %v",
+				svc.movementSensor.Name().ShortName(), detector.CameraName.ShortName(), err.Error(),
+			)
+			cameraToMovementsensor = movementsensorOrigin
+		}
+		svc.logger.Debugf("cameraToMovementsensor Pose: %v", spatialmath.PoseToProtobuf(cameraToMovementsensor.Pose()))
+
+		// determine transform from base to movement sensor
+		baseToMovementSensor, err := svc.fsService.TransformPose(ctx, movementsensorOrigin, svc.base.Name().ShortName(), nil)
+		if err != nil {
+			// here we make the assumption the movementsensor is coincident with the base
+			svc.logger.Debugf(
+				"we assume the movementsensor named: %s is coincident with the base named: %s due to err: %v",
+				svc.movementSensor.Name().ShortName(), svc.base.Name().ShortName(), err.Error(),
+			)
+			baseToMovementSensor = movementsensorOrigin
+		}
+		svc.logger.Debugf("baseToMovementSensor Pose: %v", spatialmath.PoseToProtobuf(baseToMovementSensor.Pose()))
+
+		// determine transform from base to camera
+		cameraOrigin := referenceframe.NewPoseInFrame(detector.CameraName.ShortName(), spatialmath.NewZeroPose())
+		baseToCamera, err := svc.fsService.TransformPose(ctx, cameraOrigin, svc.base.Name().ShortName(), nil)
+		if err != nil {
+			// here we make the assumption the base is coincident with the camera
+			svc.logger.Debugf(
+				"we assume the base named: %s is coincident with the camera named: %s due to err: %v",
+				svc.base.Name().ShortName(), detector.CameraName.ShortName(), err.Error(),
+			)
+			baseToCamera = cameraOrigin
+		}
+		svc.logger.Debugf("baseToCamera Pose: %v", spatialmath.PoseToProtobuf(baseToCamera.Pose()))
+
+		// get current geo position of robot
+		gp, _, err := svc.movementSensor.Position(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// instantiate a localizer and use it to get our current position
+		localizer := motion.NewMovementSensorLocalizer(svc.movementSensor, gp, spatialmath.NewZeroPose())
+		currentPIF, err := localizer.CurrentPosition(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// convert orientation of currentPIF to be left handed
+		localizerHeading := math.Mod(math.Abs(currentPIF.Pose().Orientation().OrientationVectorDegrees().Theta-360), 360)
+
+		// convert baseToMovementSensor theta to be left handed
+		localizerBaseThetaDiff := math.Mod(math.Abs(baseToMovementSensor.Pose().Orientation().OrientationVectorDegrees().Theta+360), 360)
```

Comment:
Why are we subtracting 360 from theta above & adding 360 to theta below?

Reply:
good question.
I believe the comment I have for this line is not true.

In the test case we have, our movementsensor is at (-5, 7, 0) with a 90 RH orientation wrt the base.

Hence, `baseToMovementSensor` reports `x:-4.99  y:6.99  o_z:1  theta:89.99` where theta here is a RH value.

 this line can be simplified into 
>localizerBaseThetaDiff := baseToMovementSensor.Pose().Orientation().OrientationVectorDegrees().Theta

since below we do 
> baseHeading := math.Mod(localizerHeading+localizerBaseThetaDiff, 360)

------
as an example let the movementsensor report it has a 270 LH orientation.
this means the movementsensor is facing west.
Since our `localizerBaseThetaDiff = 90` we know that the orientation of the base is 270 + 90 = 360 == 0